File: Easy/283-Move-Zeroes/Solutions.cpp
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        int kiri = 0;
        int kanan = 0;
        while (kanan < nums.size())
        {
            if (nums[kanan] != 0)
            {
                nums[kiri] = nums[kanan];
                kiri++;
            }
            kanan++;
        }
        for (kiri; kiri < nums.size(); kiri++)
        {
            nums[kiri] = 0;
        }
    }
};

File: Easy/283-Move-Zeroes/Solutions.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        kiri = 0
        for kanan in range(len(nums)):
            if nums[kanan] != 0:
                nums[kiri] = nums[kanan]
                kiri += 1
        
        while kiri < len(nums):
            nums[kiri] = 0
            kiri += 1